tests/test_forwarded.py

```diff
@@ -0,0 +1,215 @@
+from webob.middleware import forwarded
+from webob import BaseRequest, Response
+import webob.dec
+from nose.tools import assert_raises
+
+
+def test_parse_forwarded():
+    assert (forwarded.parse('for="_gazonk"') ==
+            [{'for': '_gazonk'}])
+    assert (forwarded.parse('For="[2001:db8:cafe::17]:4711"') ==
+            [{'for': '[2001:db8:cafe::17]:4711'}])
+    assert (forwarded.parse('for=192.0.2.60;proto=http;by=203.0.113.43') ==
+            [{'for': '192.0.2.60', 'proto': 'http', 'by': '203.0.113.43'}])
+    assert (forwarded.parse('for=192.0.2.43, for=198.51.100.17') ==
+            [{'for': '192.0.2.43'}, {'for': '198.51.100.17'}])
+    assert (forwarded.parse(
+        'for=192.0.2.43;by=203.0.113.43 , for=198.51.100.17') ==
+            [{'for': '192.0.2.43', 'by': '203.0.113.43'},
+             {'for': '198.51.100.17'}])
+
+
+def test_parse_forwarded_empty():
+    assert (forwarded.parse('') ==
+            [])
+
+
+def test_parse_forwarded_illegal_whitespace():
+    assert_raises(forwarded.ForwardedError,
+                  forwarded.parse, 'for =_something')
+    # a ForwardedError is a ValueError
+    assert_raises(ValueError,
+                  forwarded.parse, 'for =_something')
+    assert_raises(forwarded.ForwardedError,
+                  forwarded.parse, 'for= _something')
+    assert_raises(forwarded.ForwardedError,
+                  forwarded.parse, 'for=_something; by=192.51.100.17')
```

Comment:
Good question. Rereading the specs makes me unsure -- it has been a while
since I wrote this code. Need to study it a bit more.